### PR TITLE
attention_mask fill with -inf for UnfusedDotProductAttention

### PR DESCRIPTION
**Pull request by Agoniii**
# Description

UnfusedDotProductAttention in TE uses -10000 to fill in the attention mask, but the value is not small enough for some cases which leads to large diff between TE and HF.

The logits of model forward with FP32.
HF baseline:
```
logits: tensor([[[-0.6598, -4.5184, -4.0881,  ..., -5.0321, -5.0322, -5.0315],
         [ 2.3831,  2.2049, -0.3057,  ..., -6.5912, -6.5907, -6.5909],
         [-2.3334, -2.7213, -4.7815,  ..., -9.8526, -9.8525, -9.8523],
         ...,
         [ 3.3538,  1.4944,  1.0958,  ..., -6.4574, -6.4574, -6.4577],
         [ 5.6523,  0.4127,  1.0138,  ..., -8.7066, -8.7066, -8.7068],
         [ 0.3297, -0.7106, -0.8580,  ..., -7.7929, -7.7928, -7.7930]]],
       device='cuda:0', grad_fn=<UnsafeViewBackward0>)
```

TE before:
```
logits: tensor([[[-0.6598, -4.5184, -4.0881,  ..., -5.0321, -5.0322, -5.0316],
         [ 2.1408,  2.3935, -0.1826,  ..., -6.7005, -6.7000, -6.7002],
         [-2.5307, -3.0045, -4.9231,  ..., -9.4020, -9.4019, -9.4017],
         ...,
         [ 3.3538,  1.4944,  1.0958,  ..., -6.4574, -6.4574, -6.4578],
         [ 5.6524,  0.4127,  1.0138,  ..., -8.7066, -8.7066, -8.7068],
         [ 0.3297, -0.7106, -0.8580,  ..., -7.7929, -7.7928, -7.7930]]],
       device='cuda:0', grad_fn=<TransposeBackward0>)
```
TE after:
```
logits: tensor([[[-0.6598, -4.5184, -4.0881,  ..., -5.0321, -5.0322, -5.0316],
         [ 2.3831,  2.2049, -0.3057,  ..., -6.5912, -6.5907, -6.5909],
         [-2.3334, -2.7213, -4.7815,  ..., -9.8526, -9.8525, -9.8523],
         ...,
         [ 3.3538,  1.4944,  1.0958,  ..., -6.4574, -6.4574, -6.4578],
         [ 5.6524,  0.4127,  1.0138,  ..., -8.7066, -8.7066, -8.7068],
         [ 0.3297, -0.7106, -0.8580,  ..., -7.7929, -7.7928, -7.7930]]],
       device='cuda:0', grad_fn=<TransposeBackward0>)
```
## Type of change

- [ ] Documentation change (change only to the documentation, either a fix or a new content)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Infra/Build change
- [ ] Code refractor

## Changes

Please list the changes introduced in this PR:

- Change A
- Change B

# Checklist:

- [ ] I have read and followed the [contributing guidelines](https://github.com/NVIDIA/TransformerEngine/blob/main/CONTRIBUTING.rst)
- [ ] The functionality is complete
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by ksivaman — APPROVED** [review]
Makes sense, LGTM

File: transformer_engine/pytorch/utils.py [hunk 1]
```diff
@@ -72,7 +72,7 @@ def attention_mask_func(
     attention_scores: torch.Tensor, attention_mask: torch.Tensor
 ) -> torch.Tensor:
     """Get attention mask"""
-    attention_scores.masked_fill_(attention_mask, -10000.0)
+    attention_scores.masked_fill_(attention_mask, float("-inf"))
```
[the remark this answers is not in the document]
Comment (ksivaman) [reply transformer_engine/pytorch/utils.py:119]: @Agoniii Could you make this change as well? There's also a couple of places in the docs (`docs/examples/quickstart_utils.py` and `docs/getting_started/getting_started_utils_pytorch.py`) where we would want this change.

**Comment by pggPL** [comment]
@ptrendx please not merge it, I am unsure about this change

**Comment by pggPL** [comment]
Looked into this — I'd suggest not merging:

1. No real benefit: softmax runs in fp32, so `exp(-10000 - max)` already underflows to exactly 0 — masked positions get zero weight today.
2. `-inf` produces NaN for fully-masked rows, and the post-softmax zeroing only covers `padding*` masks, not `arbitrary`.
3. This only changes the torch fallback — the fused kernel (and Megatron-Core) still use `-10000.0` with explicit fully-masked-row handling, so results would depend on which path dispatch picks.

Making this correct would need consistent fully-masked-row handling in both paths + a test, and given (1) I don't think it's worth it. I'd close this.
